packages/extension/src/ui/features/accountTokens/SendTokenScreen.tsx

```diff
@@ -344,30 +346,51 @@ export const SendTokenScreen: FC = () => {
 
   const showSaveAddressButton = validRecipientAddress && !recipientInAddressBook
 
+  const parseInputAmount = (amount?: string, decimals?: number) => {
+    if (amount) {
+      try {
+        const result = parseAmount(amount, decimals ?? 18)
+        setIsFractionalComponentExceedsDecimals(false)
+        return result
+      } catch (e: any) {
+        if (e.message.startsWith('fractional component exceeds decimals')) {
+          setIsFractionalComponentExceedsDecimals(true)
+          console.error("e", e.mesage)
+        } else {
+          throw e
+        }
+      }
+    }
+
+    return parseAmount("0", decimals)
+  }
+
   useEffect(() => {
     if (maxClicked && maxFee && token) {
       setMaxInputAmount(token, maxFee)
     }
+
+    if (token) {
+      setParsedInputAmount(parseInputAmount(inputAmount, token.decimals))
+    }
+
     // dont add token as dependency as the reference can change
-  }, [maxClicked, maxFee, setMaxInputAmount, token?.id, token?.networkId]) // eslint-disable-line react-hooks/exhaustive-deps
+  }, [maxClicked, maxFee, setMaxInputAmount, token?.id, token?.networkId, inputAmount]) // eslint-disable-line react-hooks/exhaustive-deps
 
   if (!token) {
     return <Navigate to={routes.accounts()} />
   }
-
   const { id, name, symbol, balance, decimals, logoURI } = toTokenView(token)
 
-  const parsedInputAmount = inputAmount
-    ? parseAmount(inputAmount, decimals)
-    : parseAmount("0", decimals)
-
   const parsedTokenBalance = tokenWithBalance?.balance || parseAmount("0", decimals)
```

Comment:
`parseAmount("0", decimals)` would be 0 always?

Reply:
Updated, thanks :+1: